### DISCUSSION TITLE
Understanding Blockchain Development

### DISCUSSION THREAD

**Discussion by Csb303** [Cyfrin/foundry-full-course-cu#1106] [General]
I am currently at the end of Lesson 9 of this course coding along but I still don't feel like I understand it all fully yet at this stage, is this normal and part of the process? At what point did everything 'click' for you all?

**Comment by 0xusmanf** [comment]
Yes, this is completely normal, it took me a little while to understand what is really happening.

In my opinion, following the course and coding along will only get you so far,  you will really start to understand the topics and have a click when you start searching for answers for yourself, for example, if you don't understand how mapping works, you should search for resources about mapping, study about it on solidity docs, ask chatGPT, StackOverflow, search google.

And when you have a basic idea of how it works, start playing with it on remix to make your understanding more strong.

You can also play with the code at the end of every lesson to produce different results.

Try to solve the challenge at the end of every lesson.
